File: python/dynamic-programming/fibonacci.py


class Fibonacci():

    def __init__(self):
        self.sequence = [0,1]


    def next_number(self) -> list:
        ultimate = self.sequence[-1]
        penultimate = self.sequence[-2]

        self.sequence.append(penultimate+ultimate)

        return self.sequence
    
    def loop(self, steps: int) -> list:

        for _ in range(steps):
            self.sequence = self.next_number()

        return self.sequence
    

fib = Fibonacci()

print(fib.loop(10))

File: javascript/dynamic-programming/fibonacci.js


let arr = [0, 1]

for (let i = 1; i < 11; i++) {
    
    let ultimate = arr[arr.length - 1]
    let penultimate = arr[arr.length - 2]
    let next_number = ultimate + penultimate;

    arr.push(next_number);
}

console.log(arr)